torch/linalg/__init__.py

```diff
@@ -367,6 +367,64 @@
             [-3.1113,  2.7381]], dtype=torch.float64)
 """)
 
+householder_product = _add_docstr(_linalg.linalg_householder_product, r"""
+householder_product(input, tau, *, out=None) -> Tensor
+
+Computes the product of Householder matrices.
+
+Householder matrices are stored in compressed vector form as columns of a `(m × n)` matrix :attr:`input`,
+or columns of each matrix in a batched :attr:`input`.
+:attr:`tau` is a tensor of scalar scale factors for each Householder vector.
+For the single matrix :attr:`input` taking its `i`-th column as `vᵢ` and `τ = tau[i]`
+gives the `i`-th Householder matrix as `Hᵢ = I − τ vᵢ vᵢᴴ`.
+
+The result of this function is `H₁ H₂ ... Hᵣ`, where `r` is equal to `tau.shape[-1]`.
+This function is commonly used together with :func:`torch.geqrf`
+to explitly form the `Q` matrix of the QR decomposition.
+See `Representation of Orthogonal or Unitary Matrices`_ for further details.
+
+.. note:: LAPACK's `orgqr` is used for the computations.
```

Comment:
What about on CUDA?

Reply:
Added a note: "For CUDA inputs, cuSOLVER's `orgqr` routine is used if CUDA version >= 10.1.243."